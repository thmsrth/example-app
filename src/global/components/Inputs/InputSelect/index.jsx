import React, { Component } from 'react';

import PropTypes from 'prop-types';
import validationKeysType from '../../Form/types/validationKeys';
import validationKeysDefaults from '../../Form/defaults/validationKeys';

import InputSelect from './InputSelect';

class InputSelectWrapper extends Component {
  constructor() {
    super();
    this.state = {
      focused: false,
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleFocus = this.handleFocus.bind(this);
    this.handleBlur = this.handleBlur.bind(this);
  }

  handleChange(event) {
    const { name, value, onChange } = this.props;
    const eventValue = event ? event.value : '';
    onChange(name, typeof value === 'number'
      ? parseInt(eventValue, 10)
      : eventValue);
  }

  handleFocus(event) {
    const { onFocus } = this.props;

    this.setState({ focused: true });
    if (onFocus) onFocus(event);
  }

  handleBlur() {
    const {
      name, validationKeys, onBlur,
    } = this.props;
    const options = { type: 'select' };

    this.setState({ focused: false });
    onBlur(name, validationKeys, options);
  }

  render() {
    const {
      value,
      optionsFromDataKey,
      optionsFromData,
      hidden,
      ...otherProps
    } = this.props;
    const { focused } = this.state;

    if (hidden) return null;

    if (optionsFromDataKey) {
      return (
        <InputSelect
          {...otherProps}
          options={optionsFromData[optionsFromDataKey].data}
          loading={optionsFromData[optionsFromDataKey].loading}
          error={optionsFromData[optionsFromDataKey].error}
          value={value}
          focused={focused}
          onChange={this.handleChange}
          onFocus={this.handleFocus}
          onBlur={this.handleBlur}
        />
      );
    }

    return (
      <InputSelect
        {...otherProps}
        value={value}
        focused={focused}
        onChange={this.handleChange}
        onFocus={this.handleFocus}
        onBlur={this.handleBlur}
      />
    );
  }
}

export default InputSelectWrapper;

InputSelectWrapper.propTypes = {
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  name: PropTypes.string,
  optionsFromDataKey: PropTypes.string,
  // eslint-disable-next-line react/forbid-prop-types
  optionsFromData: PropTypes.object,
  validationKeys: validationKeysType,
  onChange: PropTypes.func,
  onFocus: PropTypes.func,
  onBlur: PropTypes.func,
  hidden: PropTypes.bool,
};

InputSelectWrapper.defaultProps = {
  value: '',
  name: '',
  optionsFromDataKey: '',
  optionsFromData: null,
  validationKeys: validationKeysDefaults,
  onChange: () => {},
  onFocus: () => {},
  onBlur: () => {},
  hidden: false,
};
