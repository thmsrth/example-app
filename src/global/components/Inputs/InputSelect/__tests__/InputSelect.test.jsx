import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import InputSelect from '../InputSelect';

const props = {
  className: 'MyInputSelect',
  name: 'MyInputName',
  label: 'My Input Label',
  options: [
    {
      value: 'first_value',
      label: 'First Value',
    },
    {
      value: 'second_value',
      label: 'Second Value',
    },
  ],
  value: 'first_value',
  disabled: false,
  focused: false,
  validationKeys: {
    required: false,
    validationAlert: '',
    warningAlert: '',
  },
  handleChange: () => {},
  handleFocus: () => {},
  handleBlur: () => {},
};

describe('InputSelect', () => {
  it('should render the component with first value', () => {
    const tree = shallow(<InputSelect {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with warning if value not in options', () => {
    const tree = shallow(<InputSelect {...props} value="" />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with second value', () => {
    const tree = shallow(<InputSelect {...props} value="second_value" />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with focused classes while focused', () => {
    const tree = shallow(<InputSelect {...props} focused />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with disabled classes while disabled', () => {
    const tree = shallow(<InputSelect {...props} disabled />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component as a required input', () => {
    const tree = shallow(<InputSelect
      {...props}
      validationKeys={{ required: true }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component as a required input', () => {
    const tree = shallow(<InputSelect
      {...props}
      validationKeys={{ required: true }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with a validation alert message', () => {
    const tree = shallow(<InputSelect
      {...props}
      validationKeys={{ validationAlert: 'My validation alert message' }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with a warning message', () => {
    const tree = shallow(<InputSelect
      {...props}
      validationKeys={{ warningAlert: 'My warning message' }}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with is loading text', () => {
    const tree = shallow(<InputSelect {...props} value="" loading="Is loading ..." />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component with is error message', () => {
    const tree = shallow(<InputSelect {...props} value="" error="Error message" />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
