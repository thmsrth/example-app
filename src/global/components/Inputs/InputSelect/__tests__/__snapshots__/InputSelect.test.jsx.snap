// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSelect should render the component as a required input 1`] = `
<div
  className="InputSelect MyInputSelect InputSelect__MyInputName MyInputSelect__MyInputName InputSelect__required MyInputSelect__required InputSelect--alert MyInputSelect--alert"
>
  <label
    className="InputSelect__label MyInputSelect__label MyInputSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label*
    <div
      className="InputSelect__alertMsg MyInputSelect__alertMsg InputSelect__MyInputName__alertMsg MyInputSelect__MyInputName__alertMsg"
    />
    <div
      className="InputSelect__input MyInputSelect__input  "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputSelect__select MyInputSelect__select InputSelect__select--alert MyInputSelect__select--alert"
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={false}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Value not in options"
        removeSelected={true}
        required={true}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value="first_value"
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputSelect__warningMsg MyInputSelect__warningMsg InputSelect__MyInputName__warningMsg MyInputSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;

exports[`InputSelect should render the component as a required input 2`] = `
<div
  className="InputSelect MyInputSelect InputSelect__MyInputName MyInputSelect__MyInputName InputSelect__required MyInputSelect__required InputSelect--alert MyInputSelect--alert"
>
  <label
    className="InputSelect__label MyInputSelect__label MyInputSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label*
    <div
      className="InputSelect__alertMsg MyInputSelect__alertMsg InputSelect__MyInputName__alertMsg MyInputSelect__MyInputName__alertMsg"
    />
    <div
      className="InputSelect__input MyInputSelect__input  "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputSelect__select MyInputSelect__select InputSelect__select--alert MyInputSelect__select--alert"
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={false}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Value not in options"
        removeSelected={true}
        required={true}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value="first_value"
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputSelect__warningMsg MyInputSelect__warningMsg InputSelect__MyInputName__warningMsg MyInputSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;

exports[`InputSelect should render the component with a validation alert message 1`] = `
<div
  className="InputSelect MyInputSelect InputSelect__MyInputName MyInputSelect__MyInputName InputSelect__required MyInputSelect__required InputSelect--alert MyInputSelect--alert"
>
  <label
    className="InputSelect__label MyInputSelect__label MyInputSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label
    <div
      className="InputSelect__alertMsg MyInputSelect__alertMsg InputSelect__MyInputName__alertMsg MyInputSelect__MyInputName__alertMsg"
    >
      My validation alert message
    </div>
    <div
      className="InputSelect__input MyInputSelect__input  "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputSelect__select MyInputSelect__select InputSelect__select--alert MyInputSelect__select--alert"
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={false}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Value not in options"
        removeSelected={true}
        required={false}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value="first_value"
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputSelect__warningMsg MyInputSelect__warningMsg InputSelect__MyInputName__warningMsg MyInputSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;

exports[`InputSelect should render the component with a warning message 1`] = `
<div
  className="InputSelect MyInputSelect InputSelect__MyInputName MyInputSelect__MyInputName InputSelect__required MyInputSelect__required InputSelect--alert MyInputSelect--alert"
>
  <label
    className="InputSelect__label MyInputSelect__label MyInputSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label
    <div
      className="InputSelect__alertMsg MyInputSelect__alertMsg InputSelect__MyInputName__alertMsg MyInputSelect__MyInputName__alertMsg"
    />
    <div
      className="InputSelect__input MyInputSelect__input  "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputSelect__select MyInputSelect__select InputSelect__select--alert MyInputSelect__select--alert"
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={false}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Value not in options"
        removeSelected={true}
        required={false}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value="first_value"
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputSelect__warningMsg MyInputSelect__warningMsg InputSelect__MyInputName__warningMsg MyInputSelect__MyInputName__warningMsg"
    >
      My warning message
    </div>
  </label>
</div>
`;

exports[`InputSelect should render the component with disabled classes while disabled 1`] = `
<div
  className="InputSelect MyInputSelect InputSelect__MyInputName MyInputSelect__MyInputName  "
>
  <label
    className="InputSelect__label MyInputSelect__label MyInputSelect__MyInputName__label InputSelect__label--disabled MyInputSelect__label--disabled MyInputSelect__MyInputName__label--disabled"
    key="MyInputName"
  >
    My Input Label
    <div
      className="InputSelect__alertMsg MyInputSelect__alertMsg InputSelect__MyInputName__alertMsg MyInputSelect__MyInputName__alertMsg"
    />
    <div
      className="InputSelect__input MyInputSelect__input  InputSelect__input--disabled MyInputSelect__input--disabled"
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputSelect__select MyInputSelect__select "
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={true}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={false}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Value not in options"
        removeSelected={true}
        required={false}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value="first_value"
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputSelect__warningMsg MyInputSelect__warningMsg InputSelect__MyInputName__warningMsg MyInputSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;

exports[`InputSelect should render the component with first value 1`] = `
<div
  className="InputSelect MyInputSelect InputSelect__MyInputName MyInputSelect__MyInputName  "
>
  <label
    className="InputSelect__label MyInputSelect__label MyInputSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label
    <div
      className="InputSelect__alertMsg MyInputSelect__alertMsg InputSelect__MyInputName__alertMsg MyInputSelect__MyInputName__alertMsg"
    />
    <div
      className="InputSelect__input MyInputSelect__input  "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputSelect__select MyInputSelect__select "
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={false}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Value not in options"
        removeSelected={true}
        required={false}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value="first_value"
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputSelect__warningMsg MyInputSelect__warningMsg InputSelect__MyInputName__warningMsg MyInputSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;

exports[`InputSelect should render the component with focused classes while focused 1`] = `
<div
  className="InputSelect MyInputSelect InputSelect__MyInputName MyInputSelect__MyInputName  "
>
  <label
    className="InputSelect__label MyInputSelect__label MyInputSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label
    <div
      className="InputSelect__alertMsg MyInputSelect__alertMsg InputSelect__MyInputName__alertMsg MyInputSelect__MyInputName__alertMsg"
    />
    <div
      className="InputSelect__input MyInputSelect__input InputSelect__input--focused MyInputSelect__input--focused "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputSelect__select MyInputSelect__select "
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={false}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Value not in options"
        removeSelected={true}
        required={false}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value="first_value"
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputSelect__warningMsg MyInputSelect__warningMsg InputSelect__MyInputName__warningMsg MyInputSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;

exports[`InputSelect should render the component with is error message 1`] = `
<div
  className="InputSelect MyInputSelect InputSelect__MyInputName MyInputSelect__MyInputName  InputSelect--alert MyInputSelect--alert"
>
  <label
    className="InputSelect__label MyInputSelect__label MyInputSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label
    <div
      className="InputSelect__alertMsg MyInputSelect__alertMsg InputSelect__MyInputName__alertMsg MyInputSelect__MyInputName__alertMsg"
    >
      Error message
    </div>
    <div
      className="InputSelect__input MyInputSelect__input  "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputSelect__select MyInputSelect__select InputSelect__select--alert MyInputSelect__select--alert"
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={false}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Nothing selected"
        removeSelected={true}
        required={false}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value=""
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputSelect__warningMsg MyInputSelect__warningMsg InputSelect__MyInputName__warningMsg MyInputSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;

exports[`InputSelect should render the component with is loading text 1`] = `
<div
  className="InputSelect MyInputSelect InputSelect__MyInputName MyInputSelect__MyInputName  "
>
  <label
    className="InputSelect__label MyInputSelect__label MyInputSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label
    <div
      className="InputSelect__alertMsg MyInputSelect__alertMsg InputSelect__MyInputName__alertMsg MyInputSelect__MyInputName__alertMsg"
    />
    <div
      className="InputSelect__input MyInputSelect__input  "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputSelect__select MyInputSelect__select "
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={true}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={false}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Is loading ..."
        removeSelected={true}
        required={false}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value=""
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputSelect__warningMsg MyInputSelect__warningMsg InputSelect__MyInputName__warningMsg MyInputSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;

exports[`InputSelect should render the component with second value 1`] = `
<div
  className="InputSelect MyInputSelect InputSelect__MyInputName MyInputSelect__MyInputName  "
>
  <label
    className="InputSelect__label MyInputSelect__label MyInputSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label
    <div
      className="InputSelect__alertMsg MyInputSelect__alertMsg InputSelect__MyInputName__alertMsg MyInputSelect__MyInputName__alertMsg"
    />
    <div
      className="InputSelect__input MyInputSelect__input  "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputSelect__select MyInputSelect__select "
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={false}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Value not in options"
        removeSelected={true}
        required={false}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value="second_value"
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputSelect__warningMsg MyInputSelect__warningMsg InputSelect__MyInputName__warningMsg MyInputSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;

exports[`InputSelect should render the component with warning if value not in options 1`] = `
<div
  className="InputSelect MyInputSelect InputSelect__MyInputName MyInputSelect__MyInputName  "
>
  <label
    className="InputSelect__label MyInputSelect__label MyInputSelect__MyInputName__label "
    key="MyInputName"
  >
    My Input Label
    <div
      className="InputSelect__alertMsg MyInputSelect__alertMsg InputSelect__MyInputName__alertMsg MyInputSelect__MyInputName__alertMsg"
    />
    <div
      className="InputSelect__input MyInputSelect__input  "
    >
      <Select
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="InputSelect__select MyInputSelect__select "
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={false}
        name="MyInputName"
        noResultsText="No results found"
        onBlur={[Function]}
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onFocus={[Function]}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={
          Array [
            Object {
              "label": "First Value",
              "value": "first_value",
            },
            Object {
              "label": "Second Value",
              "value": "second_value",
            },
          ]
        }
        pageSize={5}
        placeholder="Nothing selected"
        removeSelected={true}
        required={false}
        rtl={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        trimFilter={true}
        value=""
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="InputSelect__warningMsg MyInputSelect__warningMsg InputSelect__MyInputName__warningMsg MyInputSelect__MyInputName__warningMsg"
    />
  </label>
</div>
`;
